controllers/tokenupload_controller.go

```diff
@@ -217,35 +191,57 @@ func tryDeleteEvent(ctx context.Context, secretName string, ns string, r *TokenU
 	}
 }
 
-func findTokenByUrl(ctx context.Context, url string, ns string, r *TokenUploadReconciler, lg logr.Logger) *spi.SPIAccessToken {
+func findSpiAccessToken(ctx context.Context, uploadSecret corev1.Secret, r *TokenUploadReconciler, lg logr.Logger) (*spi.SPIAccessToken, error) {
+	spiTokenName := uploadSecret.Labels[spiTokenNameLabel]
 
-	tokenList := spi.SPIAccessTokenList{}
-	if err := r.List(ctx, &tokenList, client.InNamespace(ns)); err != nil {
-		lg.Error(err, "Can not get tokenList of tokens ")
-		return nil
+	if spiTokenName == "" {
+		lg.V(logs.DebugLevel).Info("No label found, will try to create with generated ", spiTokenNameLabel, spiTokenName)
```

Comment:
This looks like an unfinished info message? Also, this seems to inform about a logic that is then being done in another method, so maybe we should postpone the logging and do it at a place that is closer to the actual logic being applied so that there's a better chance of the two being in sync in the future :)

Reply:
OK for unfnished.
The second one not sure I understand... Do you mean it is better to have another check like if result==nil && err==nil after the function call just for putting debug log message?